src/bigistore.c

```diff
@@ -1061,6 +1059,51 @@ Datum bigistore_slice(PG_FUNCTION_ARGS)
         PG_RETURN_NULL();
 }
 
+PG_FUNCTION_INFO_V1(bigistore_slice_min_max);
+Datum bigistore_slice_min_max(PG_FUNCTION_ARGS)
+{
+    int32      min = PG_GETARG_INT32(1);
+    int32      max = PG_GETARG_INT32(2);
+    BigIStore *is  = PG_GETARG_BIGIS_COPY(0);
+
+    int min_idx = 0;
+    int i       = 0;
+    int len     = is->len;
+
+    BigIStorePair *pairs;
+    pairs = FIRST_PAIR(is, BigIStorePair);
+
+    if (min > max)
+        ereport(ERROR, (errcode(ERRCODE_SYNTAX_ERROR), errmsg("min must be less or equal max")));
+
+    if (is->len <= 0 || pairs[is->len - 1].key < min || pairs[0].key > max)
+        PG_RETURN_NULL();
+
+    if (pairs[0].key >= min && pairs[is->len - 1].key <= max)
+        PG_RETURN_POINTER(is);
+
+    for (i = 0; i < len; i++)
+    {
+        if (pairs[i].key < min)
+            min_idx = i;
+
+        if (pairs[i].key < min || pairs[i].key > max)
```

Comment:
Pairs are sorted in ascending order, therefore we should break at first sight of the element key > max.

Reply:
current assumption was that we "remove" less then we keep 
thus we deduct `is->len` and `is->buflen` whenever the key is not in range
Probably it makes sense to change the logic though